src/client/node_transformations.js

```diff
@@ -0,0 +1,95 @@
+const { binByKey } = require('./helpers/helpers.js');
+
+const binByTag = arrayToBin => binByKey('tag', arrayToBin);
+
+
+const sortWithMax = (nodeArray) => {
+  const nodesArrayCopy = nodeArray.slice(0);
+
+  nodesArrayCopy.sort((a, b) => b.avgRating - a.avgRating);
+
+  return {
+    max: nodesArrayCopy[0],
+    rest: nodesArrayCopy.slice(1),
+  };
+};
+
+const tagNodesByTag = (tagsArray, startingCx, startingCy, generateId) => {
+  const out = {};
+  const arrayCopy = tagsArray.slice(0);
+  arrayCopy.forEach((tag, i) => {
+    const newId = generateId();
+    out[tag.max.tag] = {
```

Comment:
That's confusing.. But I think right! Not sure if there's a better way of phrasing it